lucene/core/src/java/org/apache/lucene/index/IndexWriterConfig.java

```diff
@@ -478,6 +479,18 @@ public IndexWriterConfig setIndexSort(Sort sort) {
     return this;
   }
 
+  /**
+   * Set the comparator for sorting leaf readers. A DirectoryReader opened from a IndexWriter with
+   * this configuration will have its leaf readers sorted with the provided leaf sorter.
+   *
+   * @param leafSorter – a comparator for sorting leaf readers
+   * @return IndexWriterConfig with leafSorter set.
+   */
+  public IndexWriterConfig setLeafSorter(Comparator<LeafReader> leafSorter) {
```

Comment:
You added a specific unit test for this feature but we could also set a random value in `LuceneTestCase#newIndexWriterConfig` to improve the coverage.

Reply:
How will leafSorter sort leaves in this case? By what criteria?  If a sorting criteria is anything different than the existing order of leaves,  this will break some of our tests that rely on specific docIds. For example tests in the class `TestBasics` that check for specific docIDs.